### ISSUE TITLE
hi ! I have a problem

### ISSUE THREAD

**Issue by bigbrother33** [EthanRosenthal/spacecutter#4]
Actually I dont want to use skorch (more flexible), so how should i change my training code? i dont know how to add callback to my train code ..... Thanks very much !!!

**Comment by EthanRosenthal** [comment]
The callback in `spacecutter` [applies](https://pytorch.org/docs/stable/nn.html#torch.nn.Module.apply) the [clip](https://github.com/EthanRosenthal/spacecutter/blob/master/spacecutter/callbacks.py#L31) function to your pytorch module after each training batch (after you have calculated the loss and stepped the optimizer). 

You could implement the callback yourself by creating an `ascension_callback` closure and manually applying the clip function:

```python
def ascension_callback(margin=0.0, min_val=-1.0e6):

    def _clip(module):
        if isinstance(module, LogisticCumulativeLink):
            cutpoints = module.cutpoints.data
            for i in range(cutpoints.shape[0] - 1):
                cutpoints[i].clamp_(
                    min_val, cutpoints[i + 1] - margin
                )
    
    return _clip

callback = ascension_callback()

# In your training loop, do the following:

for data in data_iterator:
    # Calculate loss
    # Step optimizer
    model.apply(callback)

```

**Comment by bigbrother33** [comment]
thanks ! I have do it , and it works!!
